### PR TITLE
`fn cfl_pred`: Cleanup and make safer

### PR DESCRIPTION
**Pull request by randomPoison**
* Part of #815.
* Part of #846.

### CODE REVIEW

File: src/ipred.rs [hunk 1]
```diff
@@ -116,7 +116,7 @@ wrap_fn_ptr!(pub unsafe extern "C" fn cfl_pred(
     topleft: *const DynPixel,
     width: c_int,
     height: c_int,
-    ac: *const i16,
+    ac: &[i16; 1024],
```
Comment (kkysen) [inline src/ipred.rs:119]: Could we write these `1024`s as `32 * 32` as they are in `int16_t ac[32 * 32];` in C?

File: src/ipred.rs [hunk 2]
```diff
@@ -243,21 +243,20 @@ unsafe fn cfl_pred<BD: BitDepth>(
...
     bd: BD,
 ) {
-    let mut y = 0;
-    while y < height {
-        let mut x = 0;
-        while x < width {
-            let diff = alpha * *ac.offset(x as isize) as c_int;
-            *dst.offset(x as isize) = bd.iclip_pixel(dc + apply_sign(diff.abs() + 32 >> 6, diff));
-            x += 1;
+    let mut ac = ac.as_slice();
```
Comment (kkysen) [inline src/ipred.rs:250]:
```suggestion
    let mut ac = &ac[..width * height];
```

This should do all the bounds checks on `ac` at once and elide them elsewhere.

**Review by kkysen — APPROVED** [review]
I left a couple of small comments, but overall it LGTM.

File: src/ipred.rs (hunk 1 above)
Comment (randomPoison) [reply src/ipred.rs:119]: I'd rather introduce a constant rather than repeating the same expression everywhere, will add that to the PR before merging.

**Pull request merged**